src/metabase/api/dataset.clj

```diff
@@ -75,7 +75,12 @@
                            (assoc :metadata/model-metadata (:result_metadata source-card)))]
       (binding [qp.perms/*card-id* source-card-id]
         (qp.streaming/streaming-response [rff export-format]
-          (qp/process-query (update query :info merge info) rff))))))
+          (if was-pivot
+            (qp.pivot/run-pivot-query (assoc query
+                                             :constraints (qp.constraints/default-query-constraints)
+                                             :info        info)
```

Comment:
wouldn't it make sense to merge the info here too like you do for a regular query?